build.cake

```diff
@@ -0,0 +1,214 @@
+
+string[] SolutionList = null;
+string[] ProjList = null;
+var PROJ_EXT = "*.vbproj";
+
+//////////////////////////////////////////////////////////////////////
+// DEFINE RUN CONSTANTS
+//////////////////////////////////////////////////////////////////////
+
+var ROOT_DIR = Context.Environment.WorkingDirectory.FullPath;
+var TOOLS_DIR = ROOT_DIR + "/tools/";
+var NUNIT3_CONSOLE = TOOLS_DIR + "NUnit.ConsoleRunner/tools/nunit3-console.exe";
+var PACKAGES_CONFIG = TOOLS_DIR + "packages.config";
+
+//////////////////////////////////////////////////////////////////////
+// ERROR LOG
+//////////////////////////////////////////////////////////////////////
+
+var ErrorDetail = new List<string>();
+
+//////////////////////////////////////////////////////////////////////
+// DISCOVER SOLUTIONS
+//////////////////////////////////////////////////////////////////////
+
+Task("DiscoverSolutions")
+.Does(() =>
+    {
+        SolutionList = System.IO.Directory.GetFiles(ROOT_DIR, "*.sln", SearchOption.AllDirectories);
+        ProjList = System.IO.Directory.GetFiles(ROOT_DIR, PROJ_EXT, SearchOption.AllDirectories);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// CLEAN
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+.IsDependentOn("DiscoverSolutions")
+.Does(() =>
+    {
+        foreach(var proj in ProjList)
+            CleanDirectory(DirFrom(proj) + "/bin/" + configuration);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// RESTORE PACKAGES
+//////////////////////////////////////////////////////////////////////
+
+Task("InitializeBuild")
+.IsDependentOn("DiscoverSolutions")
+.Does(() =>
+    {
+        foreach(var sln in SolutionList)
+            NuGetRestore(sln);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("Build")
+.IsDependentOn("InitializeBuild")
+.Does(() =>
+    {
+        foreach(var proj in ProjList)
+        {
+            var sample = System.IO.Path.GetFileNameWithoutExtension(proj);
+            DisplayHeading("Building " + sample + " sample");
+
+            try
+            {
+                BuildProject(proj, configuration);
+            }
+            catch (Exception e)
+            {
+                // Just record and continue, since samples are independent
+                ErrorDetail.Add("     * " + sample + " build failed.");
+            }
+        }
+    });
+
+//////////////////////////////////////////////////////////////////////
+// RESTORE NUNIT CONSOLE
+//////////////////////////////////////////////////////////////////////
+
+Task("RestoreNUnitConsole")
+.Does(() => 
+    {
+        NuGetRestore(PACKAGES_CONFIG, 
+            new NuGetRestoreSettings()
+            {
+                PackagesDirectory = TOOLS_DIR
+            });
+    });
```

Comment:
When restoring using the bootstrap, the version number is not used for the directory. When restoring here, it looks as if if would be used. Are you making two copies when using the bootstrap?